Kernel_23/test/Kernel_23/include/CGAL/CustomKernel.h

```diff
@@ -0,0 +1,269 @@
+#include <CGAL/Cartesian.h>
+
+namespace CGAL { namespace Test {
+
+struct  Vec3 {
+  Vec3()
+  {}
+
+  Vec3(double x, double y, double z)
+   : x(x), y(y), z(z)
+    {}
+
+  Vec3(double x, double y, double z, double w)
+   : x(x/w), y(y/w), z(z/w)
+    {}
+  const double& hw() const
+  {
+      return CGAL::constant<double, 1>();
+  }
+    double x, y, z;
+};
+
+std::ostream& operator<<(std::ostream& os, const Vec3& v)
+{
+  return os << v.x << " " << v.y << " " << v.z << std::endl;
+}
+
+/// Functor that constructs a CGAL 3D bounding box given a 3D point (vector).
+template<class BaseConstructBbox>
+class ConstructBbox : public BaseConstructBbox
+{
+  public:
+    using BaseConstructBbox::operator();
+
+    /// Returns a bounding box that contains the given vector (point) \a v.
+    ///
+    /// \param v point to construct a bounding box for
+    /// \return the bounding box that contains the given point \a v
+    CGAL::Bbox_3 operator()(const Vec3 &v) const
+    {
+      return CGAL::Bbox_3(v.x, v.y, v.z, v.x, v.y, v.z);
+    }
+};
+
+/// Functor that implements various interfaces for constructing a 3D point.
+/// These need to be re-implemented to return an Motk `Vec3` instance.
+template<typename Kernel, typename CgalKernelBase>
+class ConstructPoint
+{
+  private:
+    typedef typename Kernel::RT RT;
+    typedef typename Kernel::Point_3 Point_3;
+    typedef typename Kernel::Weighted_point_3 Weighted_point_3;
+    typedef typename Point_3::Rep Rep;
+
+  public:
+
+    template<typename>
+    struct result {
+      typedef Point_3 type;
+    };
+
+    template<typename F>
+    struct result<F(Weighted_point_3)> {
+      typedef const Point_3& type;
+    };
+
+    template<typename F>
+    struct result<F(Point_3)> {
+      typedef const Point_3& type;
+    };
+
+
+    // Note : the CGAL::Return_base_tag is really internal CGAL stuff.
+    // Unfortunately it is needed for optimizing away copy-constructions,
+    // due to current lack of delegating constructors in the C++ standard.
+    /// \cond
+    Rep operator()(CGAL::Return_base_tag, CGAL::Origin o) const { return Rep(0,0,0); }
+    Rep operator()(CGAL::Return_base_tag, const RT &x, const RT &y, const RT &z) const { return Rep(x, y, z); }
+    Rep operator()(CGAL::Return_base_tag, const RT &x, const RT &y, const RT &z, const RT &w) const { return Rep(x, y, z, w); }
+    /// \endcond
+
+    /// Returns a 3D point representing the origin for the parameterized kernel.
+    ///
+    /// \return a 3D point representing the origin for the parameterized kernel
+    Point_3 operator()(const CGAL::Origin&) const { return Vec3(0, 0, 0); }
+
+    const Point_3& operator()(const Point_3& p) const { return p;}
+
+    const Point_3& operator()(const Weighted_point_3& wp) const {   std::cout << CgalKernelBase().construct_point_3_object()(wp) << std::endl; return CgalKernelBase().construct_point_3_object()(wp); }
+    /// Returns a 3D point representing the cartesian (\a x, \a y, \a z)
```

Comment:
> `` (\a x, \a y, \a z) ``

Why not

`` `(x, y, z)` ``

(also a bit further down)

Also on other places `\a` is used where it probably would better to use backticks